packages/compass-explain-plan/src/components/explain-plan-view.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useMemo, useState } from 'react';
+import {
+  Banner,
+  Icon,
+  KeylineCard,
+  SegmentedControl,
+  SegmentedControlOption,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import {
+  CodemirrorMultilineEditor,
+  prettify,
+} from '@mongodb-js/compass-editor';
+import type { ExplainPlanModalState } from '../stores/explain-plan-modal-store';
+import ExplainTree from './explain-tree';
+import { ExplainPlanSummary } from './explain-plan-side-summary';
+import ExplainCannotVisualizeBanner from './explain-cannot-visualize-banner';
+
+const viewStyles = css({
+  height: '100%',
+  display: 'grid',
+  gridTemplateColumns: '1fr',
+  gridTemplateRows: 'auto 1fr',
+  rowGap: spacing[4],
+});
+
+const viewHeaderStyles = css({
+  display: 'grid',
+  rowGap: spacing[4],
+});
+
+const viewBodyContainerStyles = css({
+  display: 'flex',
+  overflow: 'hidden',
+  gap: spacing[4],
+});
+
+const contentStyles = css({
+  flex: '1 1 0%',
+});
+
+const editorContainerStyles = css({
+  height: '100%',
+  overflow: 'hidden',
+});
+
+const editorStyles = css({
+  '& .cm-editor': {
+    paddingLeft: spacing[2],
+  },
+});
+
+const explainTreeContainerStyles = css({
+  height: '100%',
+  overflow: 'auto',
+});
+
+const summaryStyles = css({
+  flex: 'none',
+});
+
+type ExplainPlanViewProps = Partial<
+  Pick<ExplainPlanModalState, 'explainPlan' | 'rawExplainPlan' | 'error'>
+>;
+
+export const ExplainPlanView: React.FunctionComponent<ExplainPlanViewProps> = ({
+  explainPlan,
+  rawExplainPlan,
+  error,
+}) => {
+  const [viewType, setViewType] = useState<'tree' | 'json'>(
+    error ? 'json' : 'tree'
+  );
+
+  const rawExplainPlanText = useMemo(() => {
+    return rawExplainPlan
+      ? prettify(JSON.stringify(rawExplainPlan), 'json')
+      : '';
+  }, [rawExplainPlan]);
+
+  const explainPlanIndexFields = useMemo(() => {
+    return (
+      explainPlan?.usedIndexes.flatMap((index) => {
+        return Object.entries(index.fields);
+      }) ?? []
+    );
+  }, [explainPlan]);
+
+  const isParsingError = Boolean(error && !explainPlan);
+
+  if (error && !isParsingError) {
+    return <Banner variant="danger">{error}</Banner>;
+  }
+
+  return (
+    <div className={viewStyles}>
+      <div className={viewHeaderStyles}>
+        <SegmentedControl
+          size="large"
```

Comment:
first time seeing large controls in Compass, looks interesting.

Reply:
Yeah, I think it feels a bit too big when the window is small, but otherwise fits nicely with a fullscreen modal